scipy/stats/unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,869 @@
+    cdef unur_urng *urng
+    cdef object numpy_rng
+    cdef object callbacks
+    cdef object params
+
+    cdef void _set_rng(self, object seed, unur_par *par,
+                       unur_distr *distr) except *:
+        """
+        Create a UNU.RAN random number generator.
+
+        Parameters
+        ----------
+        seed : object
+            Seed for the uniform random number generator. Can be a integer,
+            NumPy BitGenerator, RandomState, or SeedSequence.
+        par : unur_par *
+            UNU.RAN parameter object.
+        distr : unur_distr *
+            UNU.RAN distribution object.
+        """
+        cdef unur_urng *urng = NULL
+
+        if seed is not None:
+            self.numpy_rng = get_numpy_rng(seed)
+            urng = get_urng(self.numpy_rng)
+            unur_set_urng(par, urng)
+
+        self.rng = unur_init(par)
+        self.urng = urng
+        unur_distr_free(distr)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef int _rvs_cont(self, double[::1] out) except -1:
+        """
+        Sample random variates from a continuous distribution.
+
+        Parameters
+        ----------
+        out : double[::1]
+            A memory view with of size ``size`` to store the result.
+        """
+        cdef:
+            ccallback_t callback
+            unuran_callback_t unur_callback
+            unur_gen *rng = self.rng
+            size_t i
+            size_t size = len(out)
+
+        init_unuran_callback(&callback, &unur_callback, self.callbacks,
+                             self.params)
+        if setjmp(callback.error_buf) != 0:
+            release_unuran_callback(&callback, &unur_callback)
+            return -1
+
+        for i in range(size):
+            out[i] = unur_sample_cont(rng)
+
+        release_unuran_callback(&callback, &unur_callback)
+
+        return 0
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef int _rvs_discr(self, int[::1] out) except -1:
+        """
+        Sample random variates from a discrete distribution.
+
+        Parameters
+        ----------
+        out : int[::1]
+            A memory view with of size ``size`` to store the result.
+        """
+        cdef:
+            ccallback_t callback
+            unuran_callback_t unur_callback
+            unur_gen *rng = self.rng
+            size_t i
+            size_t size = len(out)
+
+        init_unuran_callback(&callback, &unur_callback, self.callbacks,
+                             self.params)
+        if setjmp(callback.error_buf) != 0:
+            release_unuran_callback(&callback, &unur_callback)
+            return -1
+
+        for i in range(size):
+            out[i] = unur_sample_discr(rng)
+
+        release_unuran_callback(&callback, &unur_callback)
+
+        return 0
+
+    def rvs(self, size=None):
+        """
+        Sample from the distribution.
+
+        Parameters
+        ----------
+        size : int, optional
```

Comment:
It would be good to allow int or tuple. There is hardly any change required since you reshape the output. You just need to compute the number of elements if size is a tuple, see `rvs_ratio_uniforms`

Reply:
> It would be good to allow int or tuple

It does. Sorry, the documentation needs to be updated.

Reply:
I have updated this documentation now.

Reply:
maybe you can also add a few simple test cases to check tuple inputs

Reply:
Done.

Reply:
Marking this as resolved. Changed in bcd404ca7d2b7f0a199a37320fd6f998a10e6cf5